providers/defltprov.c

```diff
@@ -388,6 +388,9 @@ static const OSSL_ALGORITHM deflt_signature[] = {
 #ifndef OPENSSL_NO_EC
     { "ED25519:Ed25519", "provider=default", ed25519_signature_functions },
     { "ED448:Ed448", "provider=default", ed448_signature_functions },
+# ifndef OPENSSL_NO_SM2
+    { "SM2:id-ecPublicKey", "provider=default", sm2_signature_functions },
```

Comment:
Don't include id-ecPublicKey here.  That will pile it together with ECDH, and I'm pretty sure you don't want that.

Reply:
(as a matter of fact, I'm not at all sure id-ecPublicKey should be an alias for ECDH either, but that's a separate issue)

Reply:
I don't think id-ecPublicKey should be the alias.